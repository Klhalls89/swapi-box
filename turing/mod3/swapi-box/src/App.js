import React, { Component } from 'react';
import './App.scss';
import Cardholder from './components/Cardholder'

class App extends Component {
  constructor() {
    super();
      this.state = {
        movie: {},
        people: [],
        places: [],
        vehicles: []

      };
  }

  componentDidMount = () => {
    const url = 'https://swapi.co/api/films/'
    fetch(url)
      .then(response => response.json())
      .then(films => this.assignMovie(films.results))
      .catch(error => console.log(error.message))  
  }

  getPeople = () => {
    const url = 'https://swapi.co/api/people/'
    fetch(url)
      .then(response => response.json())
      .then(people => this.setPeople(people.results))
      .catch(error => console.log(error.message))  
  }

  setPeople = (people) => {
      this.setState ({
        people
      })
      this.getPeopleSpecies(people)
      this.getPeoplePlanet(people)
  }

  getPeopleSpecies = (people) => {
    console.log('getPeopleSpecies', people)
  }

  getPeoplePlanet = (people) => {
    console.log('getPeoplePlanet', people)
  }

  assignMovie = (movies) => {
    const ranMov = Math.floor(Math.random() * (7))
    const movie = movies[ranMov]
    this.setState ({
      movie
    })
  }

  render() {
   const {people} = this.state
    return (
      <div className="App">
        <header className="App-header">
          <button>Favorites</button>
          <h1 className="logo">SWAPI</h1>
          <button onClick={this.getPeople}>People</button>
          <button>Places</button>
          <button>Vehicles</button>
        </header>
        <Cardholder people={people}/> 
          <h2 className="film-name">{this.state.movie.title}</h2>
          <h2 className="release-date">{this.state.movie.release_date}</h2>
          <p className="movie-crawler">{this.state.movie.opening_crawl}</p>
      </div>
    );
  }
}

export default App;
