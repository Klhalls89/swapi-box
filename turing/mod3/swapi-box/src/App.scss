.App {
  text-align: center;
}

.App-header {
  color: white;
}

.logo {
  color: gold;
}

button {
  border-radius: 30px;
}


