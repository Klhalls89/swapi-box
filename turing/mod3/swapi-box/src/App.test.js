import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

 let mockMovie = {
  title: 'Return of the Jedi',
  release_date: '1983-05-25',
  opening_crawl: 'restore freedom to the galaxy...'
 }

describe( 'test app js', () => {
  let wrapper 

   beforeEach(() => {
    wrapper = shallow(<App />);
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe( 'get movies', () => {
    let mockFilms;

    it('should fetch movies', () => {
      mockFilms = {results:[{},{}]}
      window.fetch = jest.fn().mockImplementation(() =>  
        Promise.resolve({
          json: () => Promise.resolve({
            films: mockFilms
          })
        })
      )
      wrapper.instance().getMovies()
      expect(window.fetch).toHaveBeenCalledWith('https://swapi.co/api/films/')
    });

    it('should throw an error if fetch fails', () => {
        window.fetch = jest.fn().mockImplementation(() =>  
        Promise.reject (
          new Error('failed')
        )
      )
        wrapper.instance().getMovies()
        .then(() => {
          expect(wrapper.state('errorStatus')).toEqual('Error in fetch')
        })
    })


  })

})


