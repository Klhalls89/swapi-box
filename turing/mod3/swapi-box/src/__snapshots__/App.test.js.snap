// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app js should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
        >
          <button>
            Favorites
          </button>
          <h1
            className="logo"
          >
            SWAPI
          </h1>
          <button
            onClick={[Function]}
          >
            People
          </button>
          <button>
            Places
          </button>
          <button>
            Vehicles
          </button>
        </header>,
        <Cardholder
          people={Array []}
        />,
        <h2
          className="film-name"
        />,
        <h2
          className="release-date"
        />,
        <p
          className="movie-crawler"
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button>
              Favorites
            </button>,
            <h1
              className="logo"
            >
              SWAPI
            </h1>,
            <button
              onClick={[Function]}
            >
              People
            </button>,
            <button>
              Places
            </button>,
            <button>
              Vehicles
            </button>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorites",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SWAPI",
              "className": "logo",
            },
            "ref": null,
            "rendered": "SWAPI",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "People",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "People",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Places",
            },
            "ref": null,
            "rendered": "Places",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Vehicles",
            },
            "ref": null,
            "rendered": "Vehicles",
            "type": "button",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "people": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "film-name",
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "release-date",
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "movie-crawler",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
          >
            <button>
              Favorites
            </button>
            <h1
              className="logo"
            >
              SWAPI
            </h1>
            <button
              onClick={[Function]}
            >
              People
            </button>
            <button>
              Places
            </button>
            <button>
              Vehicles
            </button>
          </header>,
          <Cardholder
            people={Array []}
          />,
          <h2
            className="film-name"
          />,
          <h2
            className="release-date"
          />,
          <p
            className="movie-crawler"
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button>
                Favorites
              </button>,
              <h1
                className="logo"
              >
                SWAPI
              </h1>,
              <button
                onClick={[Function]}
              >
                People
              </button>,
              <button>
                Places
              </button>,
              <button>
                Vehicles
              </button>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Favorites",
              },
              "ref": null,
              "rendered": "Favorites",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SWAPI",
                "className": "logo",
              },
              "ref": null,
              "rendered": "SWAPI",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "People",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "People",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Places",
              },
              "ref": null,
              "rendered": "Places",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Vehicles",
              },
              "ref": null,
              "rendered": "Vehicles",
              "type": "button",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "people": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "film-name",
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "release-date",
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "movie-crawler",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
